src/ShellCheck/Checks/ShellSupport.hs

```diff
@@ -238,15 +242,19 @@ checkBashisms = ForShell [Sh, Dash] $ \t -> do
         warnMsg id "`<file` to read files is"
 
     bashism t@(T_SimpleCommand _ _ (cmd:arg:_))
-        | t `isCommand` "echo" && "-" `isPrefixOf` argString =
+        | t `isCommand` "echo" && hasEchoFlags =
             unless ("--" `isPrefixOf` argString) $ -- echo "-----"
                 if isDash
                 then
                     when (argString /= "-n") $
                         warnMsg (getId arg) "echo flags besides -n"
                 else
                     warnMsg (getId arg) "echo flags are"
-      where argString = concat $ oversimplify arg
+      where
+          argString = concat $ oversimplify arg
```

Comment:
You could optionally replace `concat $ oversimplify arg` with the less heavy handed version `getLiteralString arg`. This returns a `Maybe String`, with Nothing if the value is not purely literal, such as if it contains a variable (`oversimplify` instead arbitrarily inserts `"${VAR}"` in place of the variable)

Alternatively, just leave it.